CMakeLists.txt

```diff
@@ -123,7 +123,7 @@ if(NOT TARGET CURL::CURL)
       find_package(CURL REQUIRED)
     endif()
     set_target_properties(CURL::CURL PROPERTIES
-      INTERFACE_LINK_LIBRARIES "${CURL_LIBRARIES}"
+      #INTERFACE_LINK_LIBRARIES "${CURL_LIBRARIES}" 
```

Comment:
this is the only bit im worried about. im wondering if it will nuke any of our other target platforms. we can be more careful about it by checking if we are or arent on windows and doing these changes in that case. but lets see what the build systems say real quick to see

Reply:
Happy to drop that change. I can make change manually as needed.

Reply:
yeah this failed the build. i think what we can use is just a couple of `if()` `endif()`s to do different things depending on platform. let me push a commit to see if i can make CI happy and then you can let me know if it still works ok for you. does that sound ok?